pkg/scheduler/eventhandlers_test.go

```diff
@@ -334,3 +341,83 @@ func TestPreCheckForNode(t *testing.T) {
 		})
 	}
 }
+
+// test for informers of resources we care about is registered
+func TestAddAllEventHandlers(t *testing.T) {
+	tests := []struct {
+		name               string
+		gvkMap             map[framework.GVK]framework.ActionType
+		staticInformerNum  int
+		dynamicInformerNum int
+	}{
+		{
+			name:               "default handlers in framework",
+			gvkMap:             map[framework.GVK]framework.ActionType{},
+			staticInformerNum:  2,
+			dynamicInformerNum: 0,
+		},
+		{
+			name: "add GVKs handlers defined in framework dynamically",
+			gvkMap: map[framework.GVK]framework.ActionType{
+				"Pod":                               framework.Add | framework.Delete,
+				"PersistentVolume":                  framework.Delete,
+				"storage.k8s.io/CSIStorageCapacity": framework.Update,
+			},
+			staticInformerNum:  4,
+			dynamicInformerNum: 0,
+		},
+		{
+			name: "add GVKs handlers defined in plugins dynamically",
+			gvkMap: map[framework.GVK]framework.ActionType{
+				"daemonsets.v1.apps": framework.Add | framework.Delete,
+				"cronjobs.v1.batch":  framework.Delete,
+			},
+			staticInformerNum:  2,
+			dynamicInformerNum: 2,
+		},
+		{
+			name: "add GVKs handlers defined in plugins dynamically, with one illegal GVK form",
+			gvkMap: map[framework.GVK]framework.ActionType{
+				"daemonsets.v1.apps":    framework.Add | framework.Delete,
+				"custommetrics.v1beta1": framework.Update,
+			},
+			staticInformerNum:  2,
+			dynamicInformerNum: 1,
+		},
+	}
+
+	scheme := runtime.NewScheme()
+	var localSchemeBuilder = runtime.SchemeBuilder{
+		appsv1.AddToScheme,
+		batchv1.AddToScheme,
+	}
+	localSchemeBuilder.AddToScheme(scheme)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			stopCh := make(chan struct{})
+			testSched := Scheduler{
+				StopEverything:  stopCh,
+				SchedulingQueue: queue.NewTestQueue(context.Background(), nil),
+			}
+
+			client := fake.NewSimpleClientset()
+			informerFactory := informers.NewSharedInformerFactory(client, 0)
+			dynclient := dyfake.NewSimpleDynamicClient(scheme)
+			dynInformerFactory := dynamicinformer.NewDynamicSharedInformerFactory(dynclient, 0)
+
+			addAllEventHandlers(&testSched, informerFactory, dynInformerFactory, tt.gvkMap)
+
+			informerFactory.Start(testSched.StopEverything)
+			dynInformerFactory.Start(testSched.StopEverything)
+			staticInformers := informerFactory.WaitForCacheSync(testSched.StopEverything)
+			dynamicInformers := dynInformerFactory.WaitForCacheSync(testSched.StopEverything)
+
+			if tt.staticInformerNum != len(staticInformers) || tt.dynamicInformerNum != len(dynamicInformers) {
```

Comment:
Split this in 2 conditionals, one for static and one for dynamic.

Reply:
+1